RoomBookingSystem/public/js/RoomManagement.js

```diff
@@ -0,0 +1,643 @@
+            addRoomToGroupModal.appendChild(document.createElement("br"));
+
+            removeRoomFromGroupModal.appendChild(createRoomCheckbox(room));
+            removeRoomFromGroupModal.appendChild(document.createElement("br"));
+        }
+
+        grid.appendChild(card);
+    });
+}
+
+function renderGroups() {
+    const grid = document.getElementById('groupGrid');
+    const groupTemplate = document.getElementById('group-card-template');
+    const roomTemplate = groupTemplate.content.querySelector('#room-card-inside-group-card-template');
+    grid.innerHTML = ''; // Clear grid
+    grid.appendChild(groupTemplate); // Save the template again
+
+    groups.forEach(group => {
+        const groupTemplateClone = groupTemplate.content.cloneNode(true);
+
+        groupTemplateClone.querySelector('.name').textContent = group.name;
+
+        const roomGrid = groupTemplateClone.querySelector('.grid');
+        group.roomIds.forEach(roomId => {
+            const roomTemplateClone = roomTemplate.content.cloneNode(true);
+            const room = rooms.find(r => r.id === roomId);
+
+            roomTemplateClone.querySelector('.name').textContent = room.name;
+            roomTemplateClone.querySelector('.room-number').textContent = room.room_number;
+
+            roomGrid.appendChild(roomTemplateClone);
+        })
+
+        if (!notAdmin()) {
+            groupTemplateClone.querySelector('.editRoomBtns').style.display = 'block';
+
+            groupTemplateClone.querySelector('.edit-btn').onclick = () => openGroupEdit(group);
+            groupTemplateClone.querySelector('.add-btn').onclick = () => openAddRoomToGroup(group);
+            groupTemplateClone.querySelector('.remove-rooms-btn').onclick = () => openRemoveRoomsFromGroup(group);
+            groupTemplateClone.querySelector('.delete-btn').onclick = () => deleteGroup(group);
+        }
+
+        grid.appendChild(groupTemplateClone);
+    });
+}
+
+///////////////////// Room modal functions /////////////////////
+
+function closeRoomModal() {
+    document.getElementById('room-modal').style.display = 'none';
+    document.getElementById('room-name').value = '';
+    document.getElementById('room-number').value = '';
+    document.getElementById('room-max-people').value = '';
+    document.getElementById('room-description').value = '';
+}
+
+function submitRoomModal() {
+    const name = document.getElementById('room-name').value ?? '';
+    const roomNumber = document.getElementById('room-number').value ?? '';
+    const maxPeople = document.getElementById('room-max-people').value ?? '';
+    const description = document.getElementById('room-description').value ?? '';
+
+    if (notAdmin()) {
+        alert('You are not allowed to ' + window.roomModalAction + ' rooms');
+        return;
+    }
+
+    if (name === '' || roomNumber === '' || maxPeople === '' || description === '') {
+        alert('One or more of the fields are empty');
+        return;
+    }
```

Comment:
Description field should be allowed to be empty.
```suggestion
    if (name === '' || roomNumber === '' || maxPeople === '') {
        alert('One or more of the fields are empty');
        return;
    }
```